py/tests/pathfinder_worker/test_call.py

```diff
@@ -1606,6 +1606,184 @@ def test_deploy_account():
+        dummy_account_contract_definition = DeprecatedCompiledClass.Schema().loads(
+            dummy_account_contract_definition
+        )
+
+    con.execute("BEGIN")
+
+    command = EstimateFee(
+        at_block="latest",
+        chain=call.Chain.TESTNET,
+        gas_price=1,
+        pending_updates={},
+        pending_deployed=[],
+        pending_nonces={},
+        pending_timestamp=0,
+        transactions=[
+            # DECLARE an account contract class
+            DeprecatedDeclare(
+                version=0x100000000000000000000000000000000,
+                max_fee=0,
+                signature=[],
+                nonce=0,
+                contract_class=dummy_account_contract_definition,
+                sender_address=1,
+            ),
+            # DEPLOY_ACCOUNT the class declared in the previous transaction
+            DeployAccount(
+                class_hash=dummy_account_contract_class_hash,
+                contract_address_salt=0,
+                constructor_calldata=[],
+                version=0x100000000000000000000000000000001,
+                nonce=0,
+                max_fee=0,
+                signature=[],
+            ),
+        ],
+    )
+
+    (verb, output, _timings) = loop_inner(con, command)
+
+    assert output == [
+        # DECLARE an account contract class
+        {
+            "gas_consumed": 1251,
+            "gas_price": 1,
+            "overall_fee": 1251,
+        },
+        # DEPLOY_ACCOUNT the class declared in the previous transaction
+        {
+            "gas_consumed": 3096,
+            "gas_price": 1,
+            "overall_fee": 3096,
+        },
+    ]
+
+
+def test_deploy_newly_declared_sierra_account():
+    con = inmemory_with_tables()
+
+    cur = con.execute("BEGIN")
+
+    (
+        dummy_account_contract_address,
+        sierra_contract_address,
+    ) = setup_dummy_account_and_sierra_contract(cur)
+
+    con.commit()
+
+    sierra_class_definition_path = test_relative_path("./sierra_account.json.zst")
+
+    with zstandard.open(sierra_class_definition_path, "rb") as file:
+        # class_definition = file.read()
+        # class_definition = zstandard.decompress(class_definition).decode("utf-8")
+        class_definition = ContractClass.loads(file.read())
+
+    # from starkware.starknet.core.os.contract_class.class_hash import compute_class_hash
+    # class_hash = compute_class_hash(class_definition)
+    class_hash = 0x4A31654529891920D9A6F69696A23A2916B9780F830D90E452E2FA90FC9E715
+
+    # from starkware.starknet.core.os.contract_class.compiled_class_hash import (
+    #     compute_compiled_class_hash,
+    # )
+    # from starkware.starknet.services.api.contract_class.contract_class_utils import (
+    #    compile_contract_class,
+    # )
+    # compiled_class = compile_contract_class(
+    #     class_definition, allowed_libfuncs_list_name="experimental_v0.1.0"
+    # )
+    # compiled_class_hash = compute_compiled_class_hash(compiled_class)
```

Comment:
I'm not sure what the commented code is here for :detective:

Reply:
That's the code to re-compute the magic numbers below the comment. Unfortunately computing these values takes a _long_ time (especially without using the Rust Poseidon hash implementation) which adds ~20 seconds to the test runtime, so I thought I'd convert these to constants but keep the code computing the values as comments.